### ISSUE TITLE
rocm sdk 6.1.2 release

### ISSUE THREAD

**Issue by lamikr** [lamikr/rocm_sdk_builder#65]
I started porting to 6.1.2 version and it's now on wip/rocm_sdk_builder_612

Changelog so far:
- all rocm base libraries updated to rocm-6.1.2
- cmake 3.26.6 is now build on early phase after python
- added separate binfo file for pytorch_python dependencies
- dropped patches merged on upstreams
- other packages updated
	- openmpi 5.0.3
	- pytorch v.2.3.1
	- pytorch v2.3.1
	- pytorch vision v0.18.1
	- onnxruntime v1.18.0
	- deepspeed 31a57fa392 from June 7, 2024
Plan still to check at least the aotriton, ucc and ucx package updates

**Comment by jeroen-mostert** [comment]
On Manjaro (and presumably Arch) building this branch fails due to the `msgpack` dependency breaking. The fix is mentioned here: https://github.com/lamikr/rocm_sdk_builder/pull/27#issuecomment-2141882006:

> By changing line 105 of `rocm_sdk_builder/src_projects/hipBLASLt/tensilelite/Tensile/Source/lib/CMakeLists.txt` to ` find_package(msgpack REQUIRED NAMES msgpack msgpack-cxx msgpack-c)` CMake is able to find the dependency and the build can continue.

But this doesn't seem to have landed as part of the general batch of fixes for Ubuntu. 

I can't easily produce a pull at the moment as I'm spread across machines but FWIW I've attached the (unsigned) patch, main credit still goes to Dani.
[hipBLASLt-000x-Tensilelite-fix-msgpack.patch.txt](https://github.com/user-attachments/files/15793268/hipBLASLt-000x-Tensilelite-fix-msgpack.patch.txt)

**Comment by lamikr** [comment]
Thanks, I somehow originally thought that the fix was only needed to install_deps.sh. Should have read more carefully.  Based on to Danis comment, I now created the pr for this but I put @daniandtheweb as an author of the patch and you also signing it of. Is that OK.

 https://github.com/lamikr/rocm_sdk_builder/pull/76

**Comment by lamikr** [comment]
Manjaro and arch-linux patch is not merged. I also updated the hipBLASLt patches to get rid from hardcoding the install dir. Similar patch is coming to aotriton, just testing it still.

**Comment by lamikr** [comment]
First 6.1.2 release seems actually contain much more new stuff and changes that I initially thought but I think we should at least try to

- debug more the rx5700 problem showing up with pytorch-gpu-benchmark https://github.com/lamikr/rocm_sdk_builder/issues/98
- https://github.com/lamikr/rocm_sdk_builder/issues/105
- https://github.com/lamikr/rocm_sdk_builder/issues/106
- gfx1103 gpu support for https://github.com/lamikr/rocm_sdk_builder/issues/101 (7700s already work)

**Comment by jeroen-mostert** [comment]
ROCm 6.2.0 was released almost a month ago now. Of course there's no rush in this particular project, but maybe we can consider 6.1.2 "good enough" to slap a release tag on it? Some of the binfos on master have started to target 6.2.0 already, it's probably a good idea to pick a commit at some point before that and stick a pin in it.

**Comment by lamikr** [comment]
I agree, we start to be ready. I had used Fedora 40 and Mageia 9 mostly for testing and and when I did a builds on Ubuntu 22.04 and 24.04 I found some pytorch issues that needed to be fixed. 

The Ubuntu 24.04 one especially was weird, unlike on other distros, the linking failed for missing symbols in hsa-runtime until I added the search and link commands for it to cmakefiles... I don' The set command for hsa-runtime64 seemed that was in cmakefiles seemed not to be enought for ubuntu and I needed to add link command separately. t know why it's behaving differently there, cmake on ubuntu was 3.26.4... Just double testing on fedora and mageia that it did not break anything.

Then just readme.md needs to be updated for new examples and babs.sh commands.

**Comment by jeroen-mostert** [comment]
Now that you mention that, I have also had build/link issues with the 64-bit lib split on Manjaro, even with all the appropriate env variables set, but these mostly concerned third-party applications not included with the repo itself. I didn't keep track of them as I went along; eventually I got sick enough of tweaking Makefiles that I just unified `lib` and `lib64` and turned `lib64` into a symlink (the way business is normally done on Arch/Manjaro anyway).

**Comment by lamikr** [comment]
Done some release related work:
1) Tested the build again with Ubuntu 22.04, 24.04 and Fedora 40 and fixed 2 or 3 Ubuntu 22.04 related build errors that has appeared recently. There were also dependency problem in new hipTensor I had added maybe one week ago, I needed to change it to build after composable kernel.
2) ./babs.sh has been updated to include ./babs.sh --update command. I think this was important addition
    as it makes it much easier to handle updates as it will clean up all projects which has been changed so that
    they get rebuild much easier.
3) README.md has seen big update to show all changes in babs.sh functionality, added applications, changed examples, etc..
4) Changelog needs still to be done before the release
